Gem/Code/Source/UserSettings/MultiplayerSampleUserSettings.cpp

```diff
@@ -200,6 +247,135 @@ namespace MultiplayerSample
         }
     }
 
+    SpecularReflections MultiplayerSampleUserSettings::GetReflectionSetting()
+    {
+        AZ::u64 reflectionType = static_cast<AZ::u64>(DefaultReflectionType);
+
+        if (auto* registry = AZ::SettingsRegistry::Get(); registry != nullptr)
+        {
+            registry->Get(reflectionType, m_reflectionSettingKey.c_str());
+        }
+
+        return static_cast<SpecularReflections>(reflectionType);
+    }
+
+    void MultiplayerSampleUserSettings::SetReflectionSetting(SpecularReflections reflectionType)
+    {
+        if (auto* registry = AZ::SettingsRegistry::Get(); registry != nullptr)
+        {
+            // Only try to set the settings if the scene system is active.
+            // If we try to set it too early, it will assert / crash.
+            if (AzFramework::SceneSystemInterface::Get())
```

Comment:
wonder if it would be simpler to make a `m_registry` member and only set it once the dependent interfaces are ready and then replace all the `registry = AZ::SettingsRegistry::Get()` calls with `if(m_registry)`

Reply:
could potentially wait to call `MultiplayerSampleUserSettingsRequestBus` till m_registry is ready and then remove the `if(m_registry)` conditionals

Reply:
Always good to be as defensive as possible, particularly in non-performance sensitive use-cases like here. I'd prefer it if it always checked just to make sure, because that ensures the program would keep trucking along even if in a bad or inconsistent state.

Reply:
ok, so i'm a big fan of defensiveness, but i'll agree with @AMZN-alexpete that i went a little too far here, and that it's reasonable to cache the registry pointer in the constructor because the settings registry *needs* to be one of the first things started up for the engine to work correctly. It's actually a little _too_ defensive as written, because the settings would just silently fail to work correctly if the settings registry wasn't initialized. Now it will assert and crash, which is reasonable, because it would be a programming problem if this module started up in the wrong order.

The other defensive checks in here are still needed though because the settings can either get applied during boot or from the settings screen. Different subsets of systems are available depending on what point those settings calls occur.